semgrep-core/src/metachecking/Check_rule.ml

```diff
@@ -294,3 +293,153 @@ let stat_files fparser xs =
                     let s = Semgrep_prefilter_j.string_of_formula f in
                     pr2 (spf "regexp: %s" s)));
   pr2 (spf "good = %d, no regexp found = %d" !good !bad)
+
+let rec expr_to_string expr =
+  let { AST_generic.e_range; _ } = expr in
+  match e_range with
+  | Some (start, end_) ->
+      let source_file = open_in_bin start.file in
+      let extract_size = end_.charpos - start.charpos in
+      seek_in source_file start.charpos;
+      really_input_string source_file extract_size
+  | None -> failwith "invalid source/sink requires"
+
+and translate_metavar_cond cond : [> `O of (string * Yaml.value) list ] =
```

Comment:
Looks like all this code is there to support -translate_rules.
This should be put in a separate file then.